src/base/bittorrent/sessionimpl.cpp

```diff
@@ -5742,14 +5742,28 @@ void SessionImpl::handleTorrentConflictAlert(const lt::torrent_conflict_alert *a
         else
             cancelDownloadMetadata(torrentIDv1);
 
-        torrent2->nativeHandle().set_metadata(a->metadata->info_section());
+        invokeAsync([torrentHandle = torrent2->nativeHandle(), metadata = a->metadata]
+        {
+            try
+            {
+                torrentHandle.set_metadata(metadata->info_section());
+            }
+            catch (const std::exception &) {}
+        });
     }
     else if (torrent1)
     {
         if (!torrent2)
             cancelDownloadMetadata(torrentIDv2);
 
-        torrent1->nativeHandle().set_metadata(a->metadata->info_section());
+        invokeAsync([torrentHandle = torrent1->nativeHandle(), metadata = a->metadata]
+        {
+            try
```

Comment:
The libtorrent docs don't mention that [set_metadate()](http://libtorrent.org/reference-Torrent_Handle.html#set-metadata) can throw. It only mentions that it will post an alert on failure.

Reply:
> The libtorrent docs don't mention that [set_metadate()](http://libtorrent.org/reference-Torrent_Handle.html#set-metadata) can throw.

Any method of `torrent_handle` (except `is_valid()`) can throw if handle is invalid (e.g. torrent was removed). Since the qBittorrent architecture assumes tracking the life cycle of torrents, we are usually free from such a problem. But here the action is performed outside the main thread, so there is a chance that the handle was invalidated.